### DISCUSSION TITLE
bluez-alsa as default device

### DISCUSSION THREAD

**Discussion by zcomx** [Arkq/bluez-alsa#440] [Q&A]
I'm testing bluez-alsa with a bt speaker.  I tested with 'mpv --audio-device=alsa/bluealsa test.mp3' and this works.

Is there a nice way to set the blue-alsa device as the default system wide and then manually toggle back to the sound card when I disconnect from the BT speaker?

**Comment by borine** [comment]
You can easily set a bluealsa device as the system-wide default simply by providing a definition of `pcm.!default` in your ~/.asoundrc or /etc/asound.conf - just as you would for any other pcm device. And the issue of "manually toggling back" is the same as for any other pcm device too.

The ALSA libasound API has no concept of "fallback pcm", and no concept of redirecting a stream while in use. You can edit the definition of `pcm.!default` when a bluetooth device connects or disconnects, you can even do this automatically from a program or script, but the difficulty is in persuading applications that are already running to read the new definition instead of using their cached in-memory copy.

**Reply by borine** [reply]
Prompted by this question, I've written up some thoughts on using bluealsa as default pcm, and posted it to the bluez-alsa wiki here:
https://github.com/Arkq/bluez-alsa/wiki/Using-bluealsa-as-default-ALSA-PCM
I'd be interested to know if that is of any help in answering the question,
